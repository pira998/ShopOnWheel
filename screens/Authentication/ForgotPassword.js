import React from 'react';
import {
    View,
    Text,
    TouchableOpacity,
    KeyboardAvoidingView,
    Image
} from 'react-native';
import AuthLayout from './AuthLayout';
import {COLORS,SIZES,FONTS,icons} from '../../constants'


import { FormInput,CustomSwitch,AuthTextButton,TextIconButton } from '../../components';
import {utils} from '../../utils'
import {forgotPassword} from '../../firebase/api.js'
import { useFormik } from 'formik';
import * as Yup from 'yup';

import { gql, useQuery } from '@apollo/client';




const GET_USERS = gql`
    {
        users {
            id
            email
        }
    }

`

const ForgotPassword = ({navigation}) => {
    const LoginSchema = Yup.object().shape({
        email: Yup.string().email('Invalid email').required('Required'),
    });

    const {
    handleChange,
    handleBlur,
    values,
    handleSubmit,
    errors,
    touched,
    setFieldValue
  } = useFormik({
    validationSchema: LoginSchema,
    initialValues: { email: '', password: '', remember: false ,showPass:false },
    onSubmit: async (values) => {
      
        forgotPassword(values.email);
        navigation.navigate('EmailSentSuccess')
       
    }
  });




    function isEnabledSignIn(){
        return values.email != "" && typeof errors.email == 'undefined'
    }

    const {loading,error,data} = useQuery(GET_USERS,{
        onCompleted:(data) =>{
            console.log(data)
         
        }
    })


    return (
        <KeyboardAvoidingView style={{flex:1,justifyContent:"center"}} >
       <AuthLayout
        title="Password Recovery"
        subtitle="Please enter your email address to server"
        titleContainerStyle={{
            marginTop:SIZES.padding*2
        }}
       >
            {/* Form Input  */}
            {/* <Query 
                query = {GET_USERS}
            >
                {({data})=>{
                    if(typeof data.users=== undefined) return null
                    {console.log(data)}
}}

            </Query> */}
            <View
                style={{
                     flex:1,
                     marginTop:SIZES.padding*2
                }}
            >
               <FormInput 
                label="Email"
                keyboardType = "email-address"
                autoCompleteType = "email"
                onChangeText = {handleChange("email")}
                errorMsg= {errors.email}
                appendComponent={
                    <View
                        style={{
                            justifyContent:'center'
                        }}
                    >
                        <Image 
                            source={values.email==""||(values.email!=""&& typeof errors.email == 'undefined')? icons.correct:icons.cross} 
                            style = {{
                                height:20,
                                width:20,
                                tintColor:values.email==""?COLORS.gray : (values.email!=""&& typeof errors.email == 'undefined')? COLORS.green: COLORS.red
                            }}
                            />
                    </View>
                }
            
            />
             
            </View>


            {/* Button  */}
            <AuthTextButton 
                label="Send Email"
                disabled={isEnabledSignIn()?false:true}
                buttonContainerStyle={{
                    height:55,
                    alignItems:'center',
                    marginTop:SIZES.padding,
                    borderRadius: SIZES.radius,
                    backgroundColor: isEnabledSignIn()? COLORS.primary:COLORS.transparentPrimary,

                }}
                onPress={handleSubmit}
            >

            </AuthTextButton>


       
       </AuthLayout>
       </KeyboardAvoidingView>
    )
}

export default ForgotPassword;