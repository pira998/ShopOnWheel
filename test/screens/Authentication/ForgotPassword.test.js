import React from 'react';
import renderer from 'react-test-renderer';
import ForgotPassword from '.../../../screens/Authentication/ForgotPassword'
jest.mock('react-native-keyboard-aware-scroll-view', () => {
    const KeyboardAwareScrollView = ({ children }) => children;
    return { KeyboardAwareScrollView };
});

test('renders correctly', () => {
  const tree = renderer.create(<ForgotPassword />).toJSON();
  expect(tree).toMatchSnapshot();
});