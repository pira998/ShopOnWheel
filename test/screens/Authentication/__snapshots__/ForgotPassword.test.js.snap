// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingVertical": 24,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 100,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 48,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Poppins-Bold",
            "fontSize": 22,
            "lineHeight": 30,
            "textAlign": "center",
          }
        }
      >
        Password Recovery
      </Text>
      <Text
        style={
          Object {
            "color": "#525C67",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
            "lineHeight": 22,
            "marginTop": 8,
            "textAlign": "center",
          }
        }
      >
        Please enter your email address to server
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 48,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#898B9A",
                "fontFamily": "sans-serif",
                "fontSize": 14,
                "lineHeight": 22,
              }
            }
          >
            Email
          </Text>
          <Text
            style={
              Object {
                "color": "#FF1717",
                "fontFamily": "sans-serif",
                "fontSize": 14,
                "lineHeight": 22,
              }
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F8",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 55,
              "marginTop": 8,
              "paddingHorizontal": 24,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCompleteType="email"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholderTextColor="#898B9A"
            rejectResponderTermination={true}
            style={
              Object {
                "flex": 1,
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "tintColor": "#898B9A",
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(227, 120, 75, 0.4)",
          "borderRadius": 12,
          "height": 55,
          "justifyContent": "center",
          "marginTop": 24,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "sans-serif",
            "fontSize": 16,
            "lineHeight": 22,
          }
        }
      >
        Send Email
      </Text>
    </View>
  </View>
</View>
`;
